jetty-server/src/main/java/org/eclipse/jetty/server/session/Session.java

```diff
@@ -1019,6 +1013,18 @@ protected SessionData getSessionData()
     public void setResident (boolean resident)
```

Comment:
Is `setResident(true)` called for the null cache? if so are we setting timeouts that will never be used?

Reply:
The NullCache always does EVICT_ON_EXIT, but the timer would be indeed started on Session.complete() and then immediately stopped when the cache booted it out.

Fix has been applied for this to ensure that caches with EVICT_ON_EXIT never start the timer.